internal/devpkg/package.go

```diff
@@ -238,7 +238,7 @@ func (p *Package) FlakeInputName() string {
 			}
 		}
 	default:
-		result = p.installable.String() + "-" + p.Hash()
+		result = p.installable.Ref.URL + "-" + p.Hash()
```

Comment:
`Ref.URL` isn't quite the same as the flakeref in URL form. It's only set when a flakeref has the type tarball/file/git. You can use `Ref.String()` to get the ref without the attribute path:

```suggestion
		result = p.installable.Ref.String() + "-" + p.Hash()
```